synthetic_data/generators.py

```diff
@@ -47,20 +59,115 @@ def post_profile_processing_w_data(cls, data, profile):
             )
         return profile
 
-    def synthesize(self, num_samples: int, seed=None, noise_level: float = None):
+    def synthesize(
+        self,
+        num_samples: int,
+        noise_level: float = None,
+    ):
         """Generate synthetic tabular data."""
-        if seed is None:
-            seed = self.seed
-
         if noise_level is None:
             noise_level = self.noise_level
 
-        return make_data_from_report(
-            report=self.profile.report(),
-            n_samples=num_samples,
-            noise_level=noise_level,
-            seed=seed,
-        )
+        if self.is_correlated:
+            return make_data_from_report(
+                report=self.profile.report(),
+                n_samples=num_samples,
+                noise_level=noise_level,
+                seed=self.tabular_generator_seed,
+            )
+        else:
+            self.generate_column_data()
```

Comment:
call generate column data

Reply:
do you think there is a better naming here? maybe `generate_uncorrelated_data()`?

Reply:
`generate_column_data()` currently generates data from a report too ...